src/menu/option-list.js

```diff
@@ -58,3 +58,13 @@ OptionList.defaultProps = {
   overrides: {},
   resetMenu: () => {},
 };
+
+function compare(prevProps, nextProps) {
+  const highlighted = prevProps.$isHighlighted === nextProps.$isHighlighted;
+  const size = prevProps.size === nextProps.size;
+  return highlighted && size;
+}
+
+// requires updated flow parsing in the v7 branch
+// $FlowFixMe
+export default React.memo(OptionList, compare);
```

Comment:
We don't need the second param here since it does a shallow comparison by default.

Reply:
equality checks on functions are unreliable - for example on event handlers :/